src/OrchardCore/OrchardCore.Abstractions/Shell/ShellSettings.cs

```diff
@@ -48,7 +52,7 @@ public string VersionId
         public string TenantId
         {
             get => _settings["TenantId"];
-            set => _settings["TenantId"] = value;
+            init => _settings["TenantId"] = value;
```

Comment:
The init accessor is cool to make the prop read only but we may need to set it in the `UpdateShellSettingsAsync()` central place.

Reply:
It makes it read-only and also means that this value can only be set inside one of this class constructors.